tools/compiler/main.rs

```diff
@@ -74,7 +78,18 @@ fn main() -> std::io::Result<()> {
         diag.print();
         std::process::exit(-1);
     }
-    let mut compiler_config = CompilerConfiguration::new(args.format);
+
+    let mut format = args.format.clone();
+
+    if args.cpp_namespace.is_some() {
+        if args.format != format {
+            println!("C++ namespace option was set. Output format will be C++.");
```

Comment:
```suggestion
            eprintln!("C++ namespace option was set. Output format will be C++.");
```

(in case the user put the result in stdout, we don't want to pollute it with warnings)